src/scxml_sm_interface.cpp

```diff
@@ -136,6 +139,28 @@ ScxmlSMInterface::ScxmlSMInterface(const std::string& scxml_file)
   }
 }
 
+// use this to determine the next state in the state machine, given the name of the transition you'd like to query
+const QString ScxmlSMInterface::getNeighbor(const QString& state, const QString& search_text)
+{
+  QString next_state;
+
+  for (auto& pair : state_transition_map_.at(state))
+  {
+    if (pair.first == search_text)
+    {
+      return pair.second;
+    }
+  }
+  throw std::runtime_error("State '" + state.toStdString() + "' does not have a neighbor after transition '" +
+                           search_text.toStdString() + "'");
+}
+
+static bool ScxmlSMInterface::eventExists(const QString& event, const std::set<std::pair<QString, QString>>& events)
+{
+  return std::any_of(
+      events.begin(), events.end(), [&event](const std::pair<QString, QString>&) { return pair.first == event });
```

Comment:
Missing a variable name

```suggestion
      events.begin(), events.end(), [&event](const std::pair<QString, QString>& pair) { return pair.first == event });
```